internal/controller/testworkflows/testworkflow_controller.go

```diff
@@ -46,9 +53,162 @@ type TestWorkflowReconciler struct {
+		}
+
+		testWorkflow.Labels[cronjob.TestWorkflowTemplateResourceURI] = "yes"
+		return ctrl.Result{}, r.Update(ctx, &testWorkflow)
+	}
+
+	newCronJobConfigs := make(map[string]*testworkflowsv1.CronJobConfig)
+	oldCronJobs := make(map[string]*batchv1.CronJob)
+	cronJobList, err := r.CronJobClient.ListAll(ctx,
+		cronjob.GetSelector(testWorkflow.Name, cronjob.TestWorkflowResourceURI), testWorkflow.Namespace)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	for i := range cronJobList.Items {
+		oldCronJobs[cronJobList.Items[i].Spec.Schedule] = &cronJobList.Items[i]
+	}
+
+	for _, event := range events {
+		if event.Cronjob != nil {
+			if cronJob, ok := newCronJobConfigs[event.Cronjob.Cron]; !ok {
+				newCronJobConfigs[event.Cronjob.Cron] = &testworkflowsv1.CronJobConfig{
+					Cron:        event.Cronjob.Cron,
+					Labels:      event.Cronjob.Labels,
+					Annotations: event.Cronjob.Annotations,
+				}
+			} else {
+				newCronJobConfigs[cronJob.Cron] = MergeCronJobJobConfig(cronJob, event.Cronjob)
+			}
+		}
+	}
+
+	for schedule, oldCronJob := range oldCronJobs {
+		if newCronJobConfig, ok := newCronJobConfigs[schedule]; !ok {
+			// Delete removed Cron Jobs
+			if err = r.CronJobClient.Delete(ctx,
+				cronjob.GetHashedMetadataName(testWorkflow.Name, schedule), testWorkflow.Namespace); err != nil {
+				return ctrl.Result{}, err
+			}
+		} else {
+			// Update CronJob if it was created before
+			if newCronJobConfig.Labels == nil {
+				newCronJobConfig.Labels = make(map[string]string)
+			}
+
+			newCronJobConfig.Labels[cronjob.TestWorkflowResourceURI] = testWorkflow.Name
+			options := cronjob.CronJobOptions{
+				Schedule:    schedule,
+				Group:       testworkflowsv1.Group,
+				Resource:    testworkflowsv1.Resource,
+				Version:     testworkflowsv1.Version,
+				ResourceURI: cronjob.TestWorkflowResourceURI,
+				Labels:      newCronJobConfig.Labels,
+				Annotations: newCronJobConfig.Annotations,
+				Data:        "{}",
+			}
+
+			if err = r.CronJobClient.Update(ctx, oldCronJob, testWorkflow.Name,
+				cronjob.GetHashedMetadataName(testWorkflow.Name, schedule), testWorkflow.Namespace,
+				string(testWorkflow.UID), options); err != nil {
+				return ctrl.Result{}, err
+			}
+		}
+	}
+
+	for schedule, newCronJobConfig := range newCronJobConfigs {
+		if _, ok = oldCronJobs[schedule]; !ok {
+			// Create new Cron Jobs
+			if newCronJobConfig.Labels == nil {
+				newCronJobConfig.Labels = make(map[string]string)
+			}
+
+			newCronJobConfig.Labels[cronjob.TestWorkflowResourceURI] = testWorkflow.Name
+			options := cronjob.CronJobOptions{
+				Schedule:    schedule,
+				Group:       testworkflowsv1.Group,
+				Resource:    testworkflowsv1.Resource,
+				Version:     testworkflowsv1.Version,
+				ResourceURI: cronjob.TestWorkflowResourceURI,
+				Labels:      newCronJobConfig.Labels,
+				Annotations: newCronJobConfig.Annotations,
+				Data:        "[]",
+			}
+
+			if err = r.CronJobClient.Create(ctx, testWorkflow.Name,
```

Comment:
The name is unique, while it should rather allow multiple cronjobs for a single TestWorkflow - probably worth to add event index to name